pkg/sql/catalog/descs/direct.go

```diff
@@ -46,6 +46,12 @@ type Direct interface {
 		ctx context.Context, txn *kv.Txn,
 	) (nstree.Catalog, error)
 
+	// MustGetDescByID looks up a descriptor (of any type) given its ID, returning
+	// an error if the descriptor is not found.
+	MustGetDescByID(
```

Comment:
i didn't see this get used. what's this for?

Reply:
This is used in `dependentError` function to get a descriptor then do `switch` on descriptor type there.

Reply:
Can we not plumb in a collection?

Reply:
👍  dropped this method and use collection instead.